src/Maths/src/Matrix2x2.cs

```diff
@@ -0,0 +1,451 @@
+﻿using System;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+
+namespace Silk.NET.Maths
+{
+    /// <summary>A structure encapsulating a 4x4 matrix.</summary>
+    public struct Matrix2x2<T> : IEquatable<Matrix2x2<T>>
+        where T : unmanaged, IFormattable, IEquatable<T>, IComparable<T>
+    {
+        private static readonly Matrix2x2<T> _identity = new(Scalar<T>.One, Scalar<T>.Zero, Scalar<T>.Zero,
+            Scalar<T>.One);
+
+        /// <summary>Value at row 1, column 1 of the matrix.</summary>
+        public T M11;
+
+        /// <summary>Value at row 1, column 2 of the matrix.</summary>
+        public T M12;
+
+        /// <summary>Value at row 2, column 1 of the matrix.</summary>
+        public T M21;
+
+        /// <summary>Value at row 2, column 2 of the matrix.</summary>
+        public T M22;
+
+        /// <summary>Constructs a Matrix2x2 from the given components.</summary>
+        public Matrix2x2(T m11, T m12, T m21, T m22) => (M11, M12, M21, M22) = (m11, m12, m21, m22);
+
+        /// <summary>Constructs a Matrix2x2 from the given Matrix3x2.</summary>
+        /// <param name="value">The source Matrix3x2.</param>
+        public Matrix2x2(Matrix3x2<T> value)
+        {
+            M11 = value.M11;
+            M12 = value.M12;
+
+            M21 = value.M21;
+            M22 = value.M22;
+        }
+
+        /// <summary>Constructs a Matrix2x2 from the given Matrix4x3.</summary>
+        /// <param name="value">The source Matrix4x3.</param>
+        public Matrix2x2(Matrix4x3<T> value)
+        {
+            M11 = value.M11;
+            M12 = value.M12;
+
+            M21 = value.M21;
+            M22 = value.M22;
+        }
+
+        /// <summary>Constructs a Matrix2x2 from the given Matrix3x4.</summary>
+        /// <param name="value">The source Matrix3x4.</param>
+        public Matrix2x2(Matrix3x4<T> value)
+        {
+            M11 = value.M11;
+            M12 = value.M12;
+
+            M21 = value.M21;
+            M22 = value.M22;
+        }
+        
+        /// <summary>Constructs a Matrix4x4 from the given Matrix3x4.</summary>
+        /// <param name="value">The source Matrix3x4.</param>
+        public Matrix2x2(Matrix2x4<T> value)
+        {
+            M11 = value.M11;
+            M12 = value.M12;
+
+            M21 = value.M21;
+            M22 = value.M22;
+        }
+        
+        /// <summary>Constructs a Matrix4x4 from the given Matrix3x4.</summary>
+        /// <param name="value">The source Matrix3x4.</param>
+        public Matrix2x2(Matrix4x2<T> value)
+        {
+            M11 = value.M11;
+            M12 = value.M12;
+
+            M21 = value.M21;
+            M22 = value.M22;
+        }
+
+        /// <summary>Returns the multiplicative identity matrix.</summary>
+        public static Matrix2x2<T> Identity => _identity;
+
+        /// <summary>Returns whether the matrix is the identity matrix.</summary>
+        public readonly bool IsIdentity
```

Comment:
Can't be a property due to serializers getting thrown off by them.
Please convert to a method, the method version may share the same name for less typing.